### PR TITLE
[COM-352] Add Icon support to Button component

### PR DESCRIPTION
**Pull request by Angele91**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

https://impulsumvc.atlassian.net/browse/COM-352

### CODE REVIEW

File: packages/components/src/components/primitives/Button/Button.stories.tsx [hunk 1]
```diff
@@ -40,18 +41,39 @@ const AllTemplate = () => (
             </Button>
           ))}
         </HStack>
+        <HStack key={variant}>
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Button/Button.stories.tsx:44]: is this key needed?

File: packages/components/src/components/primitives/Button/Button.stories.tsx [hunk 2]
```diff
@@ -40,18 +41,39 @@ const AllTemplate = () => (
...
     ))}
   </VStack>
 );
 
 export const All = AllTemplate.bind({});
 
-const Template = (args) => <Button {...args}></Button>;
+const Template = (args) => (
+  <Button
+    {...args}
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Button/Button.stories.tsx:66]: `showLeadingIcon` and `showTrailingIcon` shouldn't be passed to the component

File: packages/components/src/theme/ComposerTheme/Components/Button.ts [hunk 3]
```diff
@@ -93,49 +93,60 @@ const generateAltButton = (color: string) => {
 
 const $spinnerSize = cssVar('spinner-size');
 
+const iconSizes = {
+  xs: '0.75rem',
+  s: '1rem',
+  m: '1.25rem',
+  l: '1.25rem',
+};
```
Comment (tkanzakic) [inline packages/components/src/theme/ComposerTheme/Components/Button.ts:101]: this has to use the Icon's size, not duplicate the values

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/components/src/components/primitives/Button/Button.stories.tsx (hunk 1 above)
Comment (Angele91) [reply packages/components/src/components/primitives/Button/Button.stories.tsx]: fixed

File: packages/components/src/components/primitives/Button/Button.stories.tsx (hunk 2 above)
Comment (Angele91) [reply packages/components/src/components/primitives/Button/Button.stories.tsx:66]: fixed

File: packages/components/src/theme/ComposerTheme/Components/Button.ts (hunk 3 above)
Comment (Angele91) [reply packages/components/src/theme/ComposerTheme/Components/Button.ts]: fixed

**Pull request closed without merging**